backup-stores/azure/src/main/java/io/camunda/zeebe/backup/azure/shared/BlobRequestsOptionsManagement.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.backup.azure.shared;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.specialized.BlobLeaseClient;
+import com.azure.storage.blob.specialized.BlobLeaseClientBuilder;
+import com.azure.storage.common.implementation.Constants;
+
+public class BlobRequestsOptionsManagement {
+  public static BlobRequestConditions acquireBlobLease(final BlobClient blobClient) {
```

Comment:
:thought_balloon: I'm not sure if this is what we want. We use this to effectively lock the blob indefinitely. This may protect us from concurrent writes to the same blob, but it does not prevent modification between creating an initial blob and then updating it. For that, we'd need to save the initial ETAG and then set it as an expectation when updating. See https://learn.microsoft.com/en-us/azure/storage/blobs/concurrency-manage

Also, what happens when we fail to release the lease? I'd imagine that this can be quite annoying as it may prevent the deletion of the container etc.